packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -231,6 +238,8 @@ export function getBeaconStateApi({
     },
 
     async postStateValidatorBalances(args, context) {
+      ensureUniqueItemsOrThrow(args.validatorIds, "Duplicate validator IDs provided");
```

Comment:
this should be enforced in `this.getStateValidatorBalances`, right now it's checking for unique items twice